robot/impl/local_robot.go

```diff
@@ -683,7 +692,14 @@ func (r *localRobot) updateWeakDependents(ctx context.Context) {
 	// For example, the framesystem should depend on all input enabled components while the web
 	// service depends on all resources.
 	// For now, we pass all resources and empty configs.
+	shouldTimeout := r.resourceConfigurationTimeout != 0
 	for resName, res := range internalResources {
+		ctx := ctx
```

Comment:
note to reviewers: I wanted to shadow `ctx` here so that we never overrode or added a timeout to the parent ctx. We could also come up with a new name instead; I opted for this approach because `ctxWithTimeout` is a lie when there is no timeout and `ctxMaybeTimeout` felt verbose and clunky. But if this approach is is non-idiomatic then I'm happy to change.